src/main/java/hudson/plugins/cobertura/adapter/CoberturaReportAdapter.java

```diff
@@ -0,0 +1,88 @@
+package hudson.plugins.cobertura.adapter;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.plugins.cobertura.CoberturaPublisher;
+import io.jenkins.plugins.coverage.adapter.CoverageReportAdapterDescriptor;
+import io.jenkins.plugins.coverage.adapter.JavaXMLCoverageReportAdapter;
+import io.jenkins.plugins.coverage.adapter.util.XMLUtils;
+import io.jenkins.plugins.coverage.detector.Detectable;
+import jenkins.model.Jenkins;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import javax.annotation.Nonnull;
+import javax.xml.transform.TransformerException;
+import java.io.File;
+
+/**
+ * Coverage report adapter for Cobertura.
+ */
+public final class CoberturaReportAdapter extends JavaXMLCoverageReportAdapter {
```

Comment:
I do not understand why this class is needed at all. Could you please clarify?

Reply:
I have moved `CoberturaReportAdapter` in Code Coverage API Plugin to test package, so we need a Cobertura Adapter implementation.

Reply:
When you say it's to test the package, do you mean it's only ever used in testing, or did I misunderstand? If it's only used for testing, maybe it could be moved into the test package

Reply:
No, I mean we have two `CoberutraReportAdapter` classes. One in API plugin and it is in test packages, I keep it is just for unit test (maybe I should remove it to avoid confusion). One in Cobertura plugin (the one we comment), it is the adapter we need to implement API plugin.